### PR TITLE
Run2-alca135 Relax ECAL energy cutoff

### PR DESCRIPTION
**Pull request by bsunanda**

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by pohsun** [comment]
+1

**Comment by fabiocos** [comment]
@bsunanda @pohsun in a private communication I understood that this PR is going to affect the size of the output in a non trivial way. Do you confirm this? Is this ok for AlCa?

**Comment by fabiocos** [comment]
@bsunanda @pohsun ping...

**Comment by bsunanda** [comment]
Yes this PR will increase the disk space requirement for AlCaReco of IsoTrackFilter by a factor between 2 & 4. On the other hand we need to study showers starting in the ECAL for a run period. We can revert it back after a months operation. Please suggest

**Comment by pohsun** [comment]
@arunhep Could you comment on this space issue?

**Comment by arunhep** [comment]
@bsunanda for what period of data taking are we talking about here? We are already quite over budget on AlCaRECOs size. And if we increase the size again by 2 to 4 times then it will increase the pressure more. This is 10_3_X and most likely we are going to use this for HI data taking of this year. 
By the way, changing the threshold from 2 GeV to 100 GeV seems quite a relaxation. Can you show us some numbers to confirm about the size?
Would you like to present the case in AlCa/DB  meeting?

**Comment by arunhep** [comment]
hold

**Comment by fabiocos** [comment]
@arunhep @bsunanda what is the situation here? Is there any update?

**Comment by fabiocos** [comment]
@arunhep @bsunanda @lpernie @pohsun ping

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by lpernie** [comment]
unhold

**Comment by arunhep** [comment]
unhold

**Comment by bsunanda** [comment]
@arunhep @lpernie Please approve this PR now

**Comment by lpernie** [comment]
+1

**Comment by fabiocos** [comment]
@lpernie @arunhep do I understand correctly that after the discussion based on the presentation https://indico.cern.ch/event/749933/contributions/3103437/attachments/1699942/2737862/IsoTrackN72.pdf by @bsunanda and the corresponding code update you have agreed on acceptable settings?

**Comment by lpernie** [comment]
Hi @fabiocos ,
I was just to send an email to summarize the outcome of the discussion.
Basically the change in rate is only needed temporarily.
There are 2 option in my options:
. Approve this PR, include it in a new release that will be used by T0. Use the release to collect data for the whole 2018E, and switch back afterward (but this seems to laborious to me).
. Create a special release with the PR. Use this release ONLY by PdmV to rereco 2018D era.
(simpler option in my opinion)

Cheers,
Luca

**Comment by bsunanda** [comment]
Hi Luca, Fabio, Arun, 
I would prefer this PR to be merged and back ported to 10_2_... (for the rereco job). The code change should remain and we change back the cuts for the final 10_3_X before it is released.
Sunanda

**Comment by lpernie** [comment]
Hello @bsunanda I will send now an email to ORP, with you in cc.
If I understand correctly this PR has 2 changes at the same time:
. change in thresholds
. change in the code
What about:
. make a new PR in 10_3 and 10_2 with only the change in the code (same thresholds)
.make an extra PR in 10_2 with only the change in thresholds, so that PdmV can rereco a single ERA with looser thresholds?
Could this work for you?

**Comment by bsunanda** [comment]
@lpernie @fabiocos OK I shall change back the thresholds in this PR and make 2 PR's for 10_2 - one which will be identical to this and the other with changed thresholds. But the second one for 10_2 has to wait till the first PR for 10_2 gets integrated. I shall make the first PR for 10_2 today

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@lpernie @arunhep I have to make an extra push. So it need to be signed again

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by lpernie** [comment]
please test

File: Calibration/HcalAlCaRecoProducers/python/alcaisotrk_cfi.py [hunk 1]
```diff
@@ -38,6 +38,5 @@
 # Prescale events only containing isolated tracks in the range
                          MomentumRangeLow  = cms.double(20.0),
                          MomentumRangeHigh = cms.double(40.0),
-                         PreScaleFactor    = cms.int32(2)
+                         PreScaleFactor    = cms.int32(10)
```
Comment (lpernie) [inline Calibration/HcalAlCaRecoProducers/python/alcaisotrk_cfi.py]: Should this PR not change anything in term of rates? why the PreScaleFactor is different?

**Review by lpernie — COMMENTED** [review]
Thanks, I have just one question.

**Comment by lpernie** [comment]
@bsunanda could you please clarify on my comment? After I can sign. Thanks

File: Calibration/HcalAlCaRecoProducers/python/alcaisotrk_cfi.py (hunk 1 above)
Comment (bsunanda) [reply Calibration/HcalAlCaRecoProducers/python/alcaisotrk_cfi.py]: We shall keep the prescale factor high for the HLT and the rate will be reduced for this by a factor of 2

**Comment by bsunanda** [comment]
@deguio @mariadalfonso @lpernie @arunhep @fabiocos Changing the thresholds with Maria's latest proposal. We shall have a factor of ~2 higher for IsoTrackFilter, a factor ~2 lower for IsoTrack (commissioning). The one for phi-symmetry will be done separately

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by mariadalfonso** [comment]
Hi @lpernie.

The idea is to have this PR in the release and not reverting it.
The same release can be used to a) take the remaining 2018 data b) used by PDMV for a special re-reco c) for the final re-reco fo all the dataset.

7% more on the HCAL should be sustainable for CMS. Is it ?
Keep in mind that this will help to provide a solution for the long standing issue of the low neutral hadrons in the endcap
Federico will put this PR in the general context of  depth1 and interdepth calibration tomorrow at the PPD.

**Comment by lpernie** [comment]
+1

**Comment by lpernie** [comment]
Ok then, let's go ahead. If there will be some need to reduce the size we can revert.

**Comment by fabiocos** [comment]
@mrodozov is cms-bot stuck? I see a "+1" from Alca but the status is not updated

**Comment by mrodozov** [comment]
@fabiocos seems to work now. sry about that, there were many cms-bot github calls stack on the queue because to that space issue :/

**Comment by fabiocos** [comment]
+1

**Pull request merged**